js/parser.js

```diff
@@ -191,11 +196,181 @@ var ERMrest = (function(module) {
                     }
                 }
 
-                context.filter = new ParsedFilter(type);
-                context.filter.setFilters(filters);
+                this._firstFilter = new ParsedFilter(type);
+                this._firstFilter.setFilters(filters);
+            }
+        }
+
+    }
+
+    Location.prototype = {
+
+        get uri() {
```

Comment:
Ideally, a client would just get the complete uri via the Location object itself. Like

```javascript
...
console.log("This reference has the following uri", reference.location);
...
```

I'm not sure how that is done, but I would expect it to work just like `window.location`. Not sure if that's a `toString()` function that gets called implicitly, but that was the idea I had in mind.